### PR TITLE
More relative coordinate space

### PR DESCRIPTION
**Pull request by smoogipooo**
This is a requisite for TimingChanges. Imagine a timing change which has start time = 5000, and a hitobject that is to be positioned inside that container with start time = 7500. Since the hit object will become relatively positioned to the container, the hitobject has to be offset backwards by the timing change start time.
This gets problematic because there is no really easy way to do this while supporting moving hit objects around in the editor.

So this allows defining the relative coordinate space as a range with a `RectangleF`. The above scenario could be achieved by:
```
new TimingPointContainer
{
    RelativeCoordinateSpace = new RectangleF(0, TimingPoint.StartTime, 0, N),
    Children = new[]
    {
        new HitObjectContainer { Y = HitObject.StartTime }
    }
}
```
Where N, in the TimingChange branch, would be calculated as the total height of all the children of the container (which is going to be a summation of HitObject start times + durations).

Am super interested at what @Tom94 thinks of this. Not sure if this is too much pollution into Drawable - if so got any better suggestions? We can't use ChildOffset here because that's applied as an absolute position.

### CODE REVIEW

File: osu.Framework/Graphics/Drawable.cs [hunk 1]
```diff
@@ -714,6 +714,28 @@ private Vector2 applyRelativeAxes(Axes relativeAxes, Vector2 v)
         /// </summary>
         private Vector2 relativeToAbsoluteFactor => Parent?.RelativeToAbsoluteFactor ?? Vector2.One;
 
+        /// <summary>
+        /// The offset to be added to relative coordinates before they're converted to absolute coordinates.
+        /// </summary>
+        private Vector2 relativePositionOffset
```
Comment (Tom94) [inline osu.Framework/Graphics/Drawable.cs]: I'd prefer this offset to be applied inline inside the `DrawPosition` getter. The name `relativePositionOffset` would imply for me that the offset is always there, regardless of `RelativePositionAxes` status.

**Comment by Tom94** [comment]
Overall I think this is fine. However, I find the representation of a coordinate system by a `RectangleF` a bit weird / hacky. Would be better if it'd just be 2 `Vector2` imho.

**Comment by smoogipooo** [comment]
What would you name it in that case? RelativeCoordinateLocation/RelativeCoordinateSize? Got a suggestion?

**Comment by Tom94** [comment]
Aaaactually, you could also make it a 3x3 matrix and allow arbitrary coordinate space shifts. Not just scale ans translation, but also shear and rotation, and then provide a helper method for setting it. Might be overkill, though. Transforming it would still work fine just as it does now, though, because matrices are linear operators.

**Comment by Tom94** [comment]
If you got the 2-vector route, `RelativeChildSize` and `RelativeChildOffset` would work fine I think.

**Comment by smoogipooo** [comment]
Shear and rotation should be a part of a child container imo. Like if you need rotated coordinates I think it's better to just make an inner relative sized container with scale set to the value you desire. I think matrix3 is very overkill.

**Pull request merged**